src/raster/rasterband.rs

```diff
@@ -863,6 +954,50 @@ pub struct StatisticsAll {
     pub std_dev: f64,
 }
 
+#[derive(Debug)]
+pub struct Histogram {
+    min: f64,
+    max: f64,
+    n_buckets: i32,
+    histogram: *mut i32,
+    histogram_vec: Option<Vec<i32>>,
```

Comment:
Is this storing the counts twice? Who owns `histogram`?

Reply:
Yes, sorry. I should document that. I used that double storage mechanisms because with `GDALGetDefaultHistogram` C owns histogram and with `GDALGetRasterHistogram` Rust does.

Reply:
No worries mate. I'm pushing those changes now.